packages/bpk-component-banner-alert/src/BpkBannerAlert.js

```diff
@@ -107,6 +107,7 @@ class BpkBannerAlert extends Component {
 
     this.state = {
       expanded: false,
+      dismissed: false,
```

Comment:
The reason we don't do this is to maintain state as high up as possible in the tree. It may make sense for us to provide a HOC which applies the default show / hide behaviour for the normal dismissable case, but we should still keep state outside of the component so that consumers of the components have full control.
If we hold `dismissed` state inside the component, for instance, it would not be possible to automatically dismiss a banner after a certain period of time has elapsed.

Reply:
But `expanded` state is inside the component, shouldn't both be in the same place, for consistency?

Having the `onDismiss` callback already allows a parent component to do some action once the component has been dismissed (maybe an `onToggle` callback should also be implemented to be able to notify parent components?)

Regarding automatically dismissing a banner after some period of time, it's funny you mention this functionality, as it's something I asked in the Slack channel and implemented but I hold the PR until the `dismissible` functionality was merged 😂  (this is the thread: https://skyscanner.slack.com/archives/C0JHPDSSU/p1509748373000119)

It's true that this functionality could be handled by a parent component, but I think it makes more sense to add a prop to this component with the period of time, and the component itself to handle it. This is my initial implementation (now I would refactor it to use `onDismiss`): https://github.com/sogame/backpack/commit/fa1dc46a7df9ab614fa956d034bcb8d69157dfc8

Maybe the best option would be to keep the `show` prop (so parent components can have more control on the visibility, if they need to), but at the same time keep the `dismissed` state in the component (as I think that most of the time, users will not want to control it). Oh, and later also add a prop to allow the component to be automatically hidden after a period of time. What do you think about this approach?

Reply:
We have a task in our backlog to move `expanded` state outside too - so ultimately they will be consistent.

I personally think that a HOC to handle state would be a better approach. Maybe something like `withNormalDismissBehaviour`. @matthewdavidson  and @k0nserv  may have some other thoughts